maskinporten-validation-core/src/main/kotlin/no/nav/pensjonsamhandling/maskinporten/validation/config/MaskinportenValidatorConfig.kt

```diff
@@ -9,11 +9,12 @@ import java.net.URL
 
 data class MaskinportenValidatorConfig(
     val baseURL: URL,
+    val jwkURL: URL? = null,
+    val issuer: String? = null,
```

Comment:
Her er det bedre å bruke default enn nullable.
`val jwkURL: URL = URL(baseUrl, "/jwk"),
val issuer: String = baseURL.toExternalForm().postfix('/'),`

Reply:
Jeg gjorde det slik da det muliggjør koden i slik den er i `MaskinportenValidatorProperties`.

Jeg ser to muligheter for å bruke default verdier.

Enten default her i `MaskinportenValidatorConfig`. Da må vi gjøre noe ala følgende i `MaskinportenValidatorProperties`.

```
    fun toConfig(): MaskinportenValidatorConfig {
        val p = proxy?.run {
            Proxy(
                Proxy.Type.HTTP,
                InetSocketAddress(
                    substringBeforeLast(':'),
                    substringAfterLast(':', "8080").toInt()
                )
            )
        }

        val b = URL(baseURL.postfix('/'))

        return when {
            jwkUrl != null && issuer != null ->
                MaskinportenValidatorConfig(baseURL = b, proxy = p)

            jwkUrl != null ->
                MaskinportenValidatorConfig(baseURL = b, jwkURL = URL(jwkUrl), proxy = p)

            issuer != null ->
                MaskinportenValidatorConfig(baseURL = b, issuer = issuer, proxy = p)

            else ->
                MaskinportenValidatorConfig(baseURL = b, jwkURL = URL(jwkUrl), issuer = issuer, proxy = p)
        }
    }
```

Eller flytte default håndteringen til `MaskinportenValidatorProperties`
```
    fun toConfig(): MaskinportenValidatorConfig {
        val base = URL(baseURL.postfix('/'))

        return MaskinportenValidatorConfig(
            baseURL = base,
            jwkURL = jwkUrl?.let { URL(it) } ?: URL(base, "/jwk"),
            issuer = issuer ?: base.toExternalForm().postfix('/'),
            proxy = proxy?.run {
                Proxy(
                    Proxy.Type.HTTP,
                    InetSocketAddress(
                        substringBeforeLast(':'),
                        substringAfterLast(':', "8080").toInt()
                    )
                )
            }
        )
    }
```

Reply:
Da tror jeg det er best å håndtere i MaskinportenValidatorProperties. Kikket på å bruke en alternativ constructor, men det ble ikke like elegant.